rule_filter/lib/rule_filter/tmail_rule.dart

```diff
@@ -14,39 +15,54 @@ part 'tmail_rule.g.dart';
 class TMailRule extends Rule {
   final RuleId? id;
   final String name;
-  final RuleCondition condition;
+  final RuleCondition? condition;
+  final RuleConditionGroup? conditionGroup;
   final RuleAction action;
 
   TMailRule({
     this.id,
     required this.name,
-    required this.condition,
+    this.condition,
+    this.conditionGroup,
     required this.action,
   });
 
   factory TMailRule.fromJson(Map<String, dynamic> json) =>
       _$TMailRuleFromJson(json);
 
-  Map<String, dynamic> toJson() => _$TMailRuleToJson(this);
+  Map<String, dynamic> toJson() {
+    final result = _$TMailRuleToJson(this);
+
+    if (result['condition'] != null) {
+      result.remove('conditionGroup');
+    } else {
+      result.remove('condition');
+    }
```

Comment:
why remove it?

Reply:
cause condition and conditionGroup is null-safety and if one of this is null, the json data parse to api will include both. It is not valid.

Reply:
why not use `includeIfNull: false`? .